PowerEditor/src/NppNotification.cpp

```diff
@@ -1074,27 +1074,28 @@ BOOL Notepad_plus::notify(SCNotification *notification)
 			_tabPopupMenu.enableItem(IDM_FILE_SAVE, isEnable);
 
 			Buffer* buf = _pEditView->getCurrentBuffer();
-			bool isUserReadOnly = buf->getUserReadOnly();
-			_tabPopupMenu.checkItem(IDM_EDIT_SETREADONLY, isUserReadOnly);
-
+			bool isDirty = buf->isDirty();
+			bool isUntitled = buf->isUntitled();
 			bool isSysReadOnly = buf->getFileReadOnly();
 			bool isInaccessible = buf->isInaccessible();
-			_tabPopupMenu.enableItem(IDM_EDIT_SETREADONLY, !isSysReadOnly && !buf->isMonitoringOn());
-			_tabPopupMenu.enableItem(IDM_EDIT_CLEARREADONLY, isSysReadOnly);
-			if (isInaccessible)
-				_tabPopupMenu.enableItem(IDM_EDIT_CLEARREADONLY, false);
-
+			bool isMonitoringOn = buf->isMonitoringOn();
+			bool isUserReadOnly = buf->getUserReadOnly();
 			bool isFileExisting = doesFileExist(buf->getFullPathName());
+
+			_tabPopupMenu.checkItem(IDM_EDIT_TOGGLE_READONLY_EDTR, isUserReadOnly);
+			_tabPopupMenu.enableItem(IDM_EDIT_TOGGLE_READONLY_EDTR, !isSysReadOnly && !isMonitoringOn);
+
+			_tabPopupMenu.checkItem(IDM_EDIT_TOGGLE_READONLY_SYS, isSysReadOnly);
+			_tabPopupMenu.enableItem(IDM_EDIT_TOGGLE_READONLY_SYS, !isInaccessible && isFileExisting && !isUntitled && !isMonitoringOn && !isDirty);
+			
 			_tabPopupMenu.enableItem(IDM_FILE_DELETE, isFileExisting);
 			_tabPopupMenu.enableItem(IDM_FILE_RELOAD, isFileExisting);
-			_tabPopupMenu.enableItem(IDM_FILE_OPEN_FOLDER, isFileExisting);
 			_tabPopupMenu.enableItem(IDM_FILE_OPEN_CMD, isFileExisting);
+			_tabPopupMenu.enableItem(IDM_FILE_OPEN_FOLDER, isFileExisting);
 			_tabPopupMenu.enableItem(IDM_FILE_CONTAININGFOLDERASWORKSPACE, isFileExisting);
 
 			_tabPopupMenu.enableItem(IDM_FILE_OPEN_DEFAULT_VIEWER, isAssoCommandExisting(buf->getFullPathName()));
-
-			bool isDirty = buf->isDirty();
-			bool isUntitled = buf->isUntitled();
+				

```

Comment:
... to this line.

Restore the original order and modify only necessary lines to make code review easily.